install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -806,6 +806,8 @@ gatewayAPI:
   enabled: false
   # -- Enable proxy protocol for all GatewayAPI listeners. Note that _only_ Proxy protocol traffic will be accepted once this is enabled.
   enableProxyProtocol: false
+  # -- Enable Backend Protocol selection support (GEP-1911) for Gateway API via appProtocol.
```

Comment:
Is there any drawback to have this enabled by default?

Reply:
@sayboras Yes, more on this here: #30452. Basically, if we enable this by default and someone depended on their downstream to choose http protocol, it will break them.

Reply:
@youngnick suggested to put it behind a flag as it's a breaking change.